src/tests-2.1/setup.ts

```diff
@@ -0,0 +1,71 @@
+import { loadDotEnv, timeout } from '../helpers';
+import { StacksCoreRpcClient } from '../core-rpc/client';
+import { PgWriteStore } from '../datastore/pg-write-store';
+import { cycleMigrations, runMigrations } from '../datastore/migrations';
+import { EventStreamServer, startEventServer } from '../event-stream/event-server';
+import { ApiServer, startApiServer } from '../api/init';
+import { ChainID } from '@stacks/transactions';
+
+export interface TestEnvContext {
+  db: PgWriteStore;
+  eventServer: EventStreamServer;
+  api: ApiServer;
+  client: StacksCoreRpcClient;
+}
+
+async function standByForPox2ToBeReady(client: StacksCoreRpcClient): Promise<void> {
```

Comment:
Just FYI I think this will have to be removed once the 2.1 branch gets merged with the latest 2.05.0.3.0 code

Reply:
You're referencing the bugged startup order right? Where the stacks-node RPC is not responsive until the API event server is responsive?

If so, this branch already has that fixed -- in this case by starting the event-server _then_ waiting for an RPC endpoint condition.